proc/threads.c

```diff
@@ -1601,12 +1613,16 @@ int proc_lockSetInterruptible(lock_t *lock)
 
 static int _proc_lockUnlock(lock_t *lock)
 {
-	thread_t *owner = lock->owner;
+	thread_t *owner = lock->owner, *current = proc_current();
```

Comment:
taking current_proc requires taking a spinlock I think it would be better to avoid calling this function in a non debug mode, as I belive that compiler won't optimize away this call.

Reply:
Fixed